src/derivative_wrappers.jl

```diff
@@ -911,14 +911,16 @@ function _jacNoise!(λ, y, p, t, S::TS, isnoise::ZygoteVJP, dgrad, dλ,
                     f(u, p, t)
                 end
             end
-            out = [back(x) for x in eachcol(Diagonal(λ))]
+            len = length(λ)
+            diag = Diagonal(λ) + 0 * reshape(repeat(y, len), len, len) # workaround: create dense diagonal matrix without scalar indexing
```

Comment:
what is this doing? It's just a conversion, so just use `adapt`?

Reply:
It's a bad hack, `adapt` seems better but I realised this entire venture is not so meaningful because there is an assumption I missed (see the newest reply in https://github.com/SciML/SciMLSensitivity.jl/issues/854) :)

Reply:
Oh, the reason is that I can't `adapt` to something that I don't know, I think? Because CUDA is not a dependency. Maybe can I adapt to `typeof(` some existing value, but I couldn't get it to work this morning

Reply:
Use `adapt(SciMLBase.parameterless_type(λ), Diagonal(λ))`